pkg/alb/loadbalancer/loadbalancer.go

```diff
@@ -547,6 +555,14 @@ func (lb *LoadBalancer) needsModification() (loadBalancerChange, bool) {
 	}
 
 	if lb.CurrentPorts != nil && lb.CurrentManagedSG != nil {
+		if lb.CurrentInboundCidrs != nil {
```

Comment:
Not 100% sure if this is required. If CurrentInboundCidrs is nil but there is a desired inbound cidr, it won't return with true.